### ISSUE TITLE
gmusic: Crash when device ID is unset

### ISSUE THREAD

**Issue by littlejeem** [beetbox/beets#3271]
Hello, I'm experiencing a bug with the gmusic plugin and so logging as requested by @sampsyo 

Setup
OS: Ubuntu 16.04.6 LTS
Python version: Python 2.7.12
beets version: 1.4.8
gmusicapi: 12

Turning off plugins made problem go away (no): In this instance its a plugin i'm trying to use (gmusic)
My configuration (output of beet config) is:

```sh
directory: /home/jlivin25/Music/Library/PlayUploads
library: /home/jlivin25/.config/beets/uploads/musiclibrary.blb
 
import:
    copy: no
    move: yes
    write: yes
    resume: ask
    quiet_fallback: skip
    timid: no
    log: beetslog.txt
ignore: .AppleDouble ._* *~ .DS_Store
art_filename: folder
 
plugins: embedart chroma fetchart convert permissions gmusic
 
pluginpath: /home/jlivin25/.config/beets/myplugins
 
threaded: yes
 
color: yes
 
match:
    strong_rec_thresh: 0.55
 
paths:
    default: $albumartist/$album/$track - $title - $artist
    singleton: Singletons/$track - $title - $artist
    comp: Various/$album/$track - $title - $artist
    albumtype:soundtrack: Soundtracks/$album/$track - $title - $artist
 
permissions:
    file: 777
    dir: 777
 
convert:
    never_convert_lossy_files: no
    copy_album_art: yes
    dest: ~/Music/Uploads
    auto: yes
    pretend: no
    embed: yes
    threads: 8
    album_art_maxwidth: 0
    quiet: no
 
    format: mp3
    formats:
        flac:
            extension: flac
            command: ffmpeg -i $source -y -vn -acodec flac $dest
        alac:
            extension: m4a
            command: ffmpeg -i $source -y -vn -acodec alac $dest
        mp3:
            extension: mp3
            command: ffmpeg -i $source -y -vn -aq 2 $dest
    max_bitrate: 500
    tmpdir:
 
gmusic:
    email: myuser@gmail.com
    password: mypassword
    auto: yes
    uploader_id:
    device_id:
    oauth_file: /home/jlivin25/.config/beets/oauth.cred
```

installed beets using and gmusicapi 12 using.

```pip install https://github.com/beetbox/beets/tarball/master```

I installed this way to take advantage of a fix to a previous bug.

```pip install gmusicapi```

The command ```beet gmusic-upload``` works correctly

The command ```beet gmusic-songs``` fails giving the following error:

```python
Traceback (most recent call last):
  File "/home/jlivin25/.local/bin/beet", line 11, in <module>
    load_entry_point('beets==1.4.9', 'console_scripts', 'beet')()
  File "/home/jlivin25/.local/lib/python2.7/site-packages/beets/ui/__init__.py", line 1266, in main
    _raw_main(args)
  File "/home/jlivin25/.local/lib/python2.7/site-packages/beets/ui/__init__.py", line 1253, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/home/jlivin25/.local/lib/python2.7/site-packages/beetsplug/gmusic.py", line 115, in search
    or Mobileclient.FROM_MAC_ADDRESS).upper()
AttributeError: 'newobject' object has no attribute 'upper'
```

**Comment by sampsyo** [comment]
Thanks! I’ll link to my proposed fix here: https://github.com/beetbox/beets/issues/3270#issuecomment-493761259

**Comment by wisp3rwind** [comment]
Closed by a fix pushed directly to master, see #3270
